### PR TITLE
ci: add prettier to pre-commit

### PR DESCRIPTION
**Pull request by barredterra**
We have prettier in frappe/frappe since almost two years. It would be nice to get clean JS files in ERPNext as well.

- Add prettier to pre-commit configuration
- Bulk-format JS + CSS files with prettier
- Ignore bulk-formatting in git blame

### CODE REVIEW

**Comment by barredterra** [comment]
Semgrep failures are not related to this PR, they just pop up because we touched the files.

**Comment by barredterra** [comment]
I'll resolve the conflicts once anybody signals willingness to merge this. Since this PR affects all JS files, conflicts will come up again and again.

**Comment by ankush** [comment]
> Since this PR affects all JS files, conflicts will come up again and again.

If we backport immediate to v14/v15 there won't be a conflict problem. I'm fine with this PR, please redo format commit after rebasing.

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+	"semi": true,
+	"tabWidth": 2,
```
Comment (ankush) [inline .prettierrc.json]: This one conflicts with editor conf. We use 4.

Comment (barredterra) [reply .prettierrc.json]: Copied this from hrms. I think it was needed for correctly formatting CSS files that were previously indented with two spaces. But let me check again.

Comment (ankush) [reply .prettierrc.json]: ERPNext doesn't have much css, we can ignore ig.

**Pull request merged**